astropy/io/ascii/mrt.py

```diff
@@ -110,8 +112,12 @@ def _set_column_val_limits(self, col):
         Sets the ``col.min`` and ``col.max`` column attributes,
         taking into account columns with Null values.
         """
-        col.max = max(col)
-        col.min = min(col)
+        try:
+            col.max = np.nanmax(col)
+            col.min = np.nanmin(col)
+        except RuntimeWarning:
+            col.max = np.nan
+            col.min = np.nan
```

Comment:
```suggestion
            warnings.warn(f"Writing column '{col.name}' of all-NaN values", UserWarning)
```
I probably would not have intercepted the warning, especially not in such a blanket way. There was [some debate](https://github.com/astropy/astropy/issues/17840#issuecomment-2697337892) on whether writing tables with such columns should be supported at all; I think as this PR will enable it, we can, but should still warn the user about it. So catching the RuntimeWarning to raise our own, a bit more informative warning is probably a good solution. No need to set `np.nan` again, since that's already what `np.nanmax` returns in this case.
You will still need the appropriate `pytest.warns()` context in the test then.